docs/interface.md

```diff
@@ -116,3 +116,34 @@ when a share is sent to a native application, the user agent may create an
 object with `ACTION_SEND`, setting the `EXTRA_SUBJECT` to `title`. Since Android
 intents do not have a URL field, `EXTRA_TEXT` would be set to `text` and `url`
 concatenated together.
+
+## Security considerations
+
+Implementations should observe the following security and privacy advice.
+
+Web Share enables data to be sent from websites to native applications. While
+this ability is not unique to Web Share, it does come with a number of potential
+security issues that may vary in severity (depending on the underlying
+platform).
+
+* Implementations MUST NOT allow the website to learn which apps are
+  installed, or which app was chosen from `navigator.share`. This information
+  could be used for fingerprinting, as well as leaking details about the user's
+  device.
+* On every call to `navigator.share`, the user MUST be presented with a dialog
+  asking them to select a target application (even if there is only one possible
+  target). This surface serves as a security confirmation, ensuring that
+  websites cannot silently send data to native applications.
+* Due to the capabilities of the API surface, websites MAY choose to restrict
+  `navigator.share` to secure browsing contexts (such as `https://` schemes).
```

Comment:
I have been deliberately avoiding going too much into spec language, as I think it makes it less understandable. We can go to that when we go into a proper spec, but for now, I want to keep it casual and readable to someone who doesn't read web specs. I'd rather just use an example here rather than linking to spec documents.